### PR TITLE
*: change source config format to YAML

### LINKED ISSUE #816: add YAML format support for source config
## Feature Request

**Is your feature request related to a problem? Please describe:**

we use YAML format for task config, but use TOML for source config now.

we can unify to use YAML for source config.

NOTE: processes config of DM-master, DM-worker and dmctl can keep to use TOML because they often deployed with DM-Ansible or TiUP.

**Describe the feature you'd like:**

**Describe alternatives you've considered:**

**Teachability, Documentation, Adoption, Migration Strategy:**

### PR DESCRIPTION
**Pull request by lance6716**
### What problem does this PR solve? 
fix #816 

### What is changed and how it works?
add `(*SourceConfig).ParseYaml`, change `operate-source` and `update-relay` using this method, keep etcd using TOML format.

### Check List 

Tests 

 - Unit test
 - Integration test
- Manual test (check print-sample-config)

Code changes

 - Has exported function/method change
 - Has interface (user interface) methods change

Side effects

 - Breaking backward compatibility

Related changes

 - Need to update the documentation
 - Need to update the `dm/dm-ansible`
 - Need to be included in the release note

### CODE REVIEW

File: dm/config/source_config.go [hunk 1]
```diff
@@ -270,8 +292,15 @@ func (c *SourceConfig) AdjustServerID(ctx context.Context, db *sql.DB) error {
...
-	return c.check(&metaData, err)
+	content, err := ioutil.ReadFile(path)
+	if err != nil {
+		return terror.ErrConfigReadCfgFromFile.Delegate(err, path)
+	}
+	if err := yaml.UnmarshalStrict(content, c); err != nil {
+		return terror.ErrConfigYamlTransform.Delegate(err, "decode source config")
+	}
+	c.adjust()
+	return nil
```
Comment (WangXiangUSTC) [inline dm/config/source_config.go:303]: how about using `ParseYaml`

Comment (lance6716) [reply dm/config/source_config.go:303]: `(*SourceConfig).check(metaData)` also checked the undecoded items, `UnmarshalStrict` is similar to that behaviour

**Pull request merged**